### DISCUSSION TITLE
About minemacs-upgrade

### DISCUSSION THREAD

**Discussion by hmanhng** [abougouffa/minemacs#165] [Q&A]
Is func `minemacs-upgrade` the same as `straight-thaw-version`?
Is the difference just that `minemacs-upgrade` will call `git-pull` first?

**Comment by abougouffa** [comment]
Basically, `minemacs-upgrade` is an async version of `minemacs-restore-locked-packages`. The latter calls `straight-x-thaw-pinned-version` among other stuff.

The `git pull` is called only if we pass a prefix argument to `minemacs-upgrade`.

**Reply by hmanhng** [reply]
ty
